### ISSUE TITLE
@at-root with dynamic selector doesn't work

### ISSUE THREAD

**Issue by chriseppstein** [sass/libsass#1297]
```scss
.test .testa {
  @at-root #{"%foo"} {
    color: red;
  }
  @extend %foo;
}
```

Compiles to:

```css
.test .testa .testa {
  color: red;
}
```

Instead of the expected:

```css
.test .testa {
  color: red;
}
```

If `@at-root #{"%foo"}` is changed to `@at-root %foo` it works as expected but these two should be semantically identical.

cc: @eoneill

**Comment by mgreter** [comment]
Duplicate of https://github.com/sass/libsass/issues/1210 and will be fixed once my latest refactoring is merged: https://github.com/sass/libsass/pull/1249!

**Comment by xzyfer** [comment]
Specs added https://github.com/sass/sass-spec/pull/419

**Comment by mgreter** [comment]
We now have a passing spec test since #1249 was merged :balloon:
